File: spring_java/src/main/java/dev/pollito/spring_java/sakila/film/domain/service/FilmUseCasesImpl.java
package dev.pollito.spring_java.sakila.film.domain.service;

import dev.pollito.spring_java.sakila.film.domain.model.Film;
import dev.pollito.spring_java.sakila.film.domain.port.in.FilmUseCases;
import dev.pollito.spring_java.sakila.film.domain.port.out.FilmRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FilmUseCasesImpl implements FilmUseCases {
  private final FilmRepository repository;

  @Override
  public Film createFilm(Film film) {
    return repository.createFilm(film);
  }

  @Override
  public Film getFilm(Integer id) {
    return repository.getFilm(id);
  }

  @Override
  public Page<Film> getFilms(Pageable pageable) {
    return repository.getFilms(pageable);
  }

  @Override
  public Film updateFilm(Integer id, Film film) {
    return repository.updateFilm(id, film);
  }

  @Override
  public void deleteFilm(Integer id) {
    repository.deleteFilm(id);
  }
}

File: spring_kotlin/src/main/kotlin/dev/pollito/spring_kotlin/sakila/film/domain/service/FilmUseCasesImpl.kt
package dev.pollito.spring_kotlin.sakila.film.domain.service

import dev.pollito.spring_kotlin.sakila.film.domain.model.Film
import dev.pollito.spring_kotlin.sakila.film.domain.port.`in`.FilmUseCases
import dev.pollito.spring_kotlin.sakila.film.domain.port.out.FilmRepository
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class FilmUseCasesImpl(private val repository: FilmRepository) : FilmUseCases {
  override fun getFilm(id: Int): Film = repository.getFilm(id)

  override fun getFilms(pageable: Pageable): Page<Film> = repository.getFilms(pageable)
}
